In [1]:
import numpy as np
import pandas as pd
import matplotlib.pyplot as plt
import seaborn as sns

In [2]:
df_acuicultura = pd.read_excel("https://www.subpesca.cl/portal/619/articles-92935_listado.xlsx")

In [3]:
#Reemplaza espacios vacíos con NA
df_acuicultura = df_acuicultura.replace(r'^\s+$', np.nan, regex=True)

/tmp/ipykernel_18801/2693579916.py:2: FutureWarning: Downcasting behavior in `replace` is deprecated and will be removed in a future version. To retain the old behavior, explicitly call `result.infer_objects(copy=False)`. To opt-in to the future behavior, set `pd.set_option('future.no_silent_downcasting', True)`
  df_acuicultura = df_acuicultura.replace(r'^\s+$', np.nan, regex=True)


In [4]:
#Convierte string en float
df_acuicultura["superficieTotal"] = pd.to_numeric(df_acuicultura["superficieTotal"], errors='coerce')

In [5]:
#Revisa información general columnas
df_acuicultura.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 3233 entries, 0 to 3232
Data columns (total 19 columns):
 #   Column                     Non-Null Count  Dtype         
---  ------                     --------------  -----         
 0   codigoCentro               3233 non-null   int64         
 1   numPert                    3233 non-null   object        
 2   Comuna                     3233 non-null   object        
 3   Region                     3233 non-null   object        
 4   Titulares                  3233 non-null   object        
 5   NumeroResSSP               3233 non-null   int64         
 6   FechaResSSP                3233 non-null   datetime64[ns]
 7   ResultadoResolucionSSP     3233 non-null   object        
 8   NumeroResSSFFAA            3233 non-null   int64         
 9   FechaResSSFFAA             3233 non-null   datetime64[ns]
 10  ResultadoResolucionSSFFAA  3233 non-null   object        
 11  EspeciesPT                 3233 non-null   object        
 12  GrupoP

In [6]:
df_acuicultura.isna().sum()

codigoCentro                    0
numPert                         0
Comuna                          0
Region                          0
Titulares                       0
NumeroResSSP                    0
FechaResSSP                     0
ResultadoResolucionSSP          0
NumeroResSSFFAA                 0
FechaResSSFFAA                  0
ResultadoResolucionSSFFAA       0
EspeciesPT                      0
GrupoPT                         9
GRUPO ESPECIE AGRUPADO          0
toponimio                       3
tipoBarrio                   1886
barrio                       1886
TipoPorcion                     5
superficieTotal                 2
dtype: int64

In [7]:
df_acuicultura.duplicated().sum()

np.int64(0)

En el caso de la columna GrupoPT faltan 9 valores. También se observa que faltan 3 valores en la columna toponimio, 1886 valores faltantes en la columna tipoBarrio, 1886 valores en la columna barrio, 5 valores en la columna TipoPorcion, y 2 valores en la columna superficieTotal. Se eliminarán las columnas **tipoBarrio** y **barrio** debido a que contienen una gran cantidad de valores nulos, mientras que en los otros casos se revisará si se pueden completar los datos faltantes o si sólo es posible eliminar las filas que contengan valores nulos.

No se detectaron filas duplicadas en este dataset.


In [8]:
#Elimina columnas con gran cantidad de valores nulos
df_acuicultura = df_acuicultura.drop(['tipoBarrio', 'barrio'], axis=1)

In [9]:
df_acuicultura_NA = df_acuicultura[df_acuicultura.isna().any(axis=1)]

In [10]:
df_acuicultura_NA

codigoCentro    numPert                          Comuna  \
8            10058  210012005  IQUIQUE                          
148          30176  207031011  CALDERA                          
149          30177  207031014  CALDERA                          
264          90157  199091033  CARAHUE                          
476         104214  208103064  QUINCHAO                         
493          30170  207031018  CALDERA                          
497          30169  207031013  CALDERA                          
630         110778  201111788  AISEN                            
674          30163  205031028  CALDERA                          
1398         40162   94043013  LOS VILOS                        
3080        100024  900000037  CALBUCO                          
3201        100285  900000026  CORRAL                           
3205         90017  900000009  TOLTEN                           
3220         30002  900000057  CHAÑARAL                         

                                                 Region  \
8                                    REGIÓN DE TARAPACA   
148                                   REGIÓN DE ATACAMA   
149                                   REGIÓN DE ATACAMA   
264                              REGIÓN DE LA ARAUCANÍA   
476                                 REGIÓN DE LOS LAGOS   
493                                   REGIÓN DE ATACAMA   
497                                   REGIÓN DE ATACAMA   
630   REGIÓN DE AYSÉN DEL GENERAL CARLOS IBÁÑEZ DEL ...   
674                                   REGIÓN DE ATACAMA   
1398                                 REGIÓN DE COQUIMBO   
3080                                REGIÓN DE LOS LAGOS   
3201                                 REGIÓN DE LOS RÍOS   
3205                             REGIÓN DE LA ARAUCANÍA   
3220                                  REGIÓN DE ATACAMA   

                                         Titulares  NumeroResSSP FechaResSSP  \
8                77424630-4: MAINSTREAM CHILE S.A.          2426  2022-11-23   
148                 79732410-8: HIDROCULTIVOS S.A.          2871  2014-10-28   
149                 79732410-8: HIDROCULTIVOS S.A.          2785  2014-10-21   
264     11775443-K: SEPULVEDA ALVAREZ, JUAN CARLOS          1884  2014-07-22   
476   4756758-0: HERNANDEZ VALENZUELA, JOSE MIGUEL          2570  2010-08-23   
493                 79732410-8: HIDROCULTIVOS S.A.          1186  2010-03-30   
497                 79732410-8: HIDROCULTIVOS S.A.           813  2010-02-18   
630                 76003885-7: AUSTRALIS MAR S.A.          3242  2007-11-13   
674                    93711000-6: CAMANCHACA S.A.           402  2008-02-08   
1398    4146491-7: SPAARWATER GILDEMEISTER, JOHANN           551  1997-04-17   
3080      7928655-9: MIRANDA VELASQUEZ, JUAN PEDRO           426  1983-11-21   
3201       5723757-0: VILLANUEVA ARRIAGADA, GERMAN            53  1983-02-10   
3205           5826927-1: JARAMILLO POZAS, ALBERTO           413  1980-10-21   
3220    79510340-6: CULTIVOS MARINOS FLAMENCO S.A.           366  1984-08-01   

     ResultadoResolucionSSP  NumeroResSSFFAA FechaResSSFFAA  \
8                   Aprueba             1604     2023-06-30   
148                 Aprueba              357     2016-01-26   
149                 Aprueba             3305     2015-12-23   
264                 Aprueba             3246     2014-09-17   
476                 Aprueba             2076     2011-06-16   
493                 Aprueba             2132     2010-12-24   
497                 Aprueba             2004     2010-12-06   
630                 Aprueba             2287     2009-11-24   
674                 Aprueba             1694     2008-12-30   
1398                Aprueba             1127     2004-04-19   
3080                Aprueba              333     1990-05-07   
3201                Aprueba              386     1987-05-05   
3205                Aprueba              151     1987-02-06   
3220                Aprueba             1013     1984-12-14  

Se procede a completar datos de la columna **GrupoPT**, dado que es posible deducir los grupos faltantes a partir de las columna **EspeciePT**. Esto fue realizado para minimizar la pérdida
de datos al limpiar el dataset.

Además, se cambia el **GRUPO ESPECIE AGRUPADO** del erizo rojo a EQUINODERMOS, porque clasificarlo como OTROS no entrega información relevante acerca de esta especie.

In [11]:
df_acuicultura.at[8,'GrupoPT']='CIPRÍNIDOS'

In [12]:
df_acuicultura.at[148,'GrupoPT']='MACROALGAS'

In [13]:
df_acuicultura.at[149,'GrupoPT']='MACROALGAS'

In [14]:
df_acuicultura.at[493,'GrupoPT']='MACROALGAS'

In [15]:
df_acuicultura.at[497,'GrupoPT']='MACROALGAS'

In [16]:
df_acuicultura.at[674,'GrupoPT']='MACROALGAS'

In [17]:
df_acuicultura.at[3220,'GrupoPT']='MACROALGAS'

In [18]:
df_acuicultura.at[476,'GrupoPT']='VENÉRIDOS'

In [19]:
df_acuicultura.at[1398,'GrupoPT']='EQUINOIDES'

In [20]:
df_acuicultura.at[1398,'GRUPO ESPECIE AGRUPADO']='EQUINODERMOS'

In [21]:
df_acuicultura_NA = df_acuicultura[df_acuicultura.isna().any(axis=1)]
df_acuicultura_NA

codigoCentro    numPert                          Comuna  \
264          90157  199091033  CARAHUE                          
630         110778  201111788  AISEN                            
3080        100024  900000037  CALBUCO                          
3201        100285  900000026  CORRAL                           
3205         90017  900000009  TOLTEN                           

                                                 Region  \
264                              REGIÓN DE LA ARAUCANÍA   
630   REGIÓN DE AYSÉN DEL GENERAL CARLOS IBÁÑEZ DEL ...   
3080                                REGIÓN DE LOS LAGOS   
3201                                 REGIÓN DE LOS RÍOS   
3205                             REGIÓN DE LA ARAUCANÍA   

                                       Titulares  NumeroResSSP FechaResSSP  \
264   11775443-K: SEPULVEDA ALVAREZ, JUAN CARLOS          1884  2014-07-22   
630               76003885-7: AUSTRALIS MAR S.A.          3242  2007-11-13   
3080    7928655-9: MIRANDA VELASQUEZ, JUAN PEDRO           426  1983-11-21   
3201     5723757-0: VILLANUEVA ARRIAGADA, GERMAN            53  1983-02-10   
3205         5826927-1: JARAMILLO POZAS, ALBERTO           413  1980-10-21   

     ResultadoResolucionSSP  NumeroResSSFFAA FechaResSSFFAA  \
264                 Aprueba             3246     2014-09-17   
630                 Aprueba             2287     2009-11-24   
3080                Aprueba              333     1990-05-07   
3201                Aprueba              386     1987-05-05   
3205                Aprueba              151     1987-02-06   

     ResultadoResolucionSSFFAA  \
264                    Aprueba   
630                    Aprueba   
3080                   Aprueba   
3201                   Aprueba   
3205                   Aprueba   

                                             EspeciesPT     GrupoPT  \
264                              CHOLGA, CHORITO, CHORO   MITILIDOS   
630   SALMON CEREZA, SALMON DEL ATLANTICO, SALMON KE...  SALMONIDOS   
3080                             CHOLGA, CHORITO, CHORO   MITILIDOS   
3201                             CHOLGA, CHORITO, CHORO   MITILIDOS   
3205                             CHOLGA, CHORITO, CHORO   MITILIDOS   

     GRUPO ESPECIE AGRUPADO                                      toponimio  \
264                MOLUSCOS    (1): RÍO IMPERIAL, AL SUROESTE DE NEHUENTUÉ   
630                   PECES  (1): ISLAS BLANCAS, AL ESTE DE ISLA MC INTYRE   
3080               MOLUSCOS                                            NaN   
3201               MOLUSCOS                                            NaN   
3205               MOLUSCOS                                            NaN   

     TipoPorcion  superficieTotal  
264          NaN             1.12  
630          NaN             3.00  
3080         NaN              NaN  
3201         NaN              NaN  
3205         NaN             0.00

In [22]:
df_acuicultura = df_acuicultura.dropna()

In [23]:
df_acuicultura.isna().sum()

codigoCentro                 0
numPert                      0
Comuna                       0
Region                       0
Titulares                    0
NumeroResSSP                 0
FechaResSSP                  0
ResultadoResolucionSSP       0
NumeroResSSFFAA              0
FechaResSSFFAA               0
ResultadoResolucionSSFFAA    0
EspeciesPT                   0
GrupoPT                      0
GRUPO ESPECIE AGRUPADO       0
toponimio                    0
TipoPorcion                  0
superficieTotal              0
dtype: int64

No se observan valores nulos en este dataset.